### PR TITLE
Add TypeScript typings

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/tschaub/es-main/issues/9

### CODE REVIEW

**Review by remcohaszing — APPROVED** [review]
LGTM

**Review by micalevisk — APPROVED** [review]
LGTM as well

**Review by micalevisk — CHANGES_REQUESTED** [review]
actually could you please add the type defs for this new function?

https://github.com/tschaub/es-main/blob/8a569c0ca2345221e62a1dedeb00f9a29301178f/main.js#L5-L12

**Comment by cinderblock** [comment]
```ts
export default function (meta: ImportMeta): boolean;
export function stripExt(name: String): String;
```

**Review by micalevisk — CHANGES_REQUESTED** [review]
please add this

File: main.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export default function (meta: ImportMeta): boolean;
```
Comment (micalevisk) [inline main.d.ts:1]:
```suggestion
export default function (meta: ImportMeta): boolean;
export function stripExt(name: string): string;
```

**Review by q-garethallenstringer — APPROVED** [review]

**Comment by cinderblock** [comment]
We could also add JSDoc comments to the published `main.d.ts` to annotate the function with a little more detail in code editors.

**Comment by hasparus** [comment]
Hey @tschaub, I'd hate to annoy you, but is there a chance of merging this in the future?
Do you need some help getting it past the finish line, or would you prefer the typings to live in DefinitelyTyped?

**Comment by tschaub** [comment]
Thanks for the proposed contribution @sidvishnoi and to the others here for the reviews and suggestions.

In #24, JSDoc comments were added with the types and documentation for the exported functions.  The `main.d.ts` and `main.d.ts.map` files are generated from these comments.  The 1.1.1 release includes these, so I hope things should work better for TypeScript users now.

**Pull request closed without merging**